packages/models/src/classMap.js

```diff
@@ -81,9 +57,27 @@ export default class ClassMap {
   }
 
   codeObjectFromEvent(event) {
-    return this.codeObjects.find(
-      (co) => co.id.indexOf(event.toString()) !== -1
-    );
+    let codeObject;
+    // These types of events should not be reporting path and lineno, but sometimes
+    // they do.
+    if (!(event.httpServerRequest || event.httpClientRequest || event.sql)) {
+      const { path, lineno } = event;
+      const location = [path, lineno].filter((e) => e).join(':');
+      if (location !== '') {
+        const codeObjects = this.codeObjectsAtLocation(location);
```

Comment:
This is the way that the agent spec expects events and code objects to be correlated.